test/integration/detachable_joint.cc

```diff
@@ -213,3 +213,137 @@ TEST_F(DetachableJointTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(LinksInSameModel))
   // the expected distance.
   EXPECT_GT(b2Poses.front().Pos().Z() - b2Poses.back().Pos().Z(), expDist);
 }
+
+ /////////////////////////////////////////////////
+ // Test for re-attaching a detached joint. This uses the vehicle_blue and B1
+ // box models. The B1 model is first detached from the vehicle. Although
+ // detached, the distance (x-direction) between B1 and vehicle is 1.5, which
+ // is the default offset. Then, linear velocity of 1.0 is published on the
+ // `/cmd_vel` topic. After 200 iterations, the B1 model will remain in the same
+ // position whereas the vehicle will move in the x-direction. Now the
+ // distance between B1 and the vehicle will be the default offset (1.5)
+ // in addition to the distance traveled by the vehicle. Next, B1 is re-attached
+ // to the vehicle. After 200 iterations, we can confirm that B1 has moved with
+ // the vehicle and the distance traveled by B1 is close to that of the vehicle.
+ // Therefore, it confirms that B1 is re-attached to the vehicle.
+ TEST_F(DetachableJointTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(ReAttach))
+ {
+   using namespace std::chrono_literals;
+
+   this->StartServer("/test/worlds/detachable_joint.sdf");
```

Comment:
`common::joinPaths`

Reply:
sorry what is this `common::joinPaths`?

Reply:
https://github.com/gazebosim/gz-common/blob/7e04a830d6573c5bb76ae018a5d3da31707081c2/include/ignition/common/Filesystem.hh#L148-L150

Rather than using raw paths, you should use:

```
this->StartServer(common::joinPaths("test", "worlds", "detachable_joint.sdf"));
```

Not every platform (windows notably) uses `/` for a path separater, and the `joinPaths` function will do the right thing in all cases.

Reply:
addressed

Reply:
addressed [here](https://github.com/gazebosim/gz-sim/pull/1687/files#diff-62dd49d28ff49e56afa8f3787907a979acee6497f90d001fe1656e71004a5ccdL66-R66) and [here](https://github.com/gazebosim/gz-sim/pull/1687/files#diff-62dd49d28ff49e56afa8f3787907a979acee6497f90d001fe1656e71004a5ccdL144-R144)

Reply:
wait I forgot to run the test before pushing the code. I realized that taking out the `/` results in the following error. Therefore, I think we should keep the `/`
```
[Err] [SystemPaths.cc:482] File [/home/liam/gz_fortress/src/gz-simtest/worlds/detachable_joint.sdf] resolved to path [/home/liam/gz_fortress/src/gz-simtest/worlds/detachable_joint.sdf] but the path does not exist
[Err] [Server.cc:109] Failed to find world [/home/liam/gz_fortress/src/gz-simtest/worlds/detachable_joint.sdf]
```

Reply:
@azeey in case you didn't see above

Reply:
I saw it. I think the `StartServer` function also needs to be updated to use `jointPaths`, but I didn't want to block this PR on that. It could be done in a follow-up PR. Your workaround should work for now.